effekt/shared/src/main/scala/effekt/generator/llvm/Transformer.scala

```diff
@@ -548,25 +544,31 @@ object Transformer {
 
   def shareValue(value: machine.Variable)(using FunctionContext, BlockContext): Unit = {
     value.tpe match {
+      case machine.Type.Extern(name) => {
+        val decl = name.split(" ")
+        if (decl.length > 2) {
```

Comment:
This looks very ad hoc :) What is happening here?

Reply:
It kind of is. An `extern type` gets a `String` body, which can (will) be backend-specific. IIUC, for LLVM, there are 4 kinds of information we need: 
1. The LLVM 'type' (e.g. `%Int`)
2. The Number of bits it occupies
3. ~ How to increase refcounts
4. ~ How to decrease refcounts

3 and 4 are optional (only needed for types that are partially on the heap).

Yes, ideally we would use a more structured format here.

Reply:
I think it would be good to have a function called "parseExternSomething" (or variations thereof) that explains this in a comment.

Reply:
Agreed. I have changed that now.